### ISSUE TITLE
Carbon Modal component accessibility violation message "Content is not within a landmark element"

### ISSUE THREAD

**Issue by rsingh54** [carbon-design-system/carbon#7675]
**Detailed description : -**

Getting accessibility violation message -  ```Content is not within a landmark element``` while using the carbon modal component.

![image](https://user-images.githubusercontent.com/49453089/106046743-a6918f80-60b0-11eb-9452-44d0a83695bf.png)

![image](https://user-images.githubusercontent.com/49453089/106045581-20287e00-60af-11eb-98e7-3148e3618577.png)

```
Versions of Carbon Design System:

"carbon-components": "10.11.1",
"carbon-components-react": "7.11.1",
"carbon-icons": "7.0.7",
"react": "^16.13.1",
"react-dom": "^16.13.1",

```

**Comment by tw15egan** [comment]
I believe this is because you need to add a `role` to one of the containing elements, see the WAIA Aria guidelines for landmarks and choose the one appropriate for your application: [More information](https://www.w3.org/TR/wai-aria-practices/examples/landmarks/main.html)

The a11y violation does not occur in ur storybook because it is contained inside of a `<div role="main">`

<img width="1121" alt="Screen Shot 2021-01-27 at 12 10 11 PM" src="https://user-images.githubusercontent.com/11928039/106047572-a1284b00-6098-11eb-826d-051f51af8656.png">

Hope this helps!

**Comment by rsingh54** [comment]
I can see the role `presentation` already added to the containing elements from carbon.However, I don't see the role `presentation` in the list of landmarks here:


![image](https://user-images.githubusercontent.com/49453089/106164068-984a7e80-6157-11eb-818d-d5393c27b85e.png)


As can be seen here, after changing the role to `complementary` along with` aria-labelledby="compl2"` and div id of `compl2` in the child span element, the accessibility violation is resolved.

![image](https://user-images.githubusercontent.com/49453089/106164313-d051c180-6157-11eb-985b-450abf7ba322.png)

**`Additional Info`**

Also, I tried using the roles `Main` and `Contentinfo `with or without `aria-labelledby` and it resolves the accessibility violation `Content is not within a landmark element`. Ideally we should be using roles with lowercase such as` "main"` and `contentinfo.`

**With `aria-labelledby` and role `Main`**


![image](https://user-images.githubusercontent.com/49453089/106165791-7520ce80-6159-11eb-926b-610db65e38db.png)


**Without `aria-labelledby` and role `Main`**

![image](https://user-images.githubusercontent.com/49453089/106165648-4f93c500-6159-11eb-8a7c-f992e59a0954.png)

Since the role ```presentation``` already exists, there shouldn't be any violation.Also, we are not sure why the role presentation is not listed in the `ARIA Landmarks List`.
We need a fix for this.We don't have a control over the roles and aria-labelledby.

**Comment by tw15egan** [comment]
Can you create a reduced test case on CodeSandbox so we can properly debug the issue? The `role="main"` on the main container div is what should resolve the issue, not the `role="presentation"`

**Comment by tw15egan** [comment]
Going to close this out because it is working as intended when placed inside a proper region, like in our Storybook. If you feel like there is more for us to do, please feel free to reopen this with a link to a reproducible example so we can get this sorted out. Thanks!

**Comment by rsingh54** [comment]
> Can you create a reduced test case on CodeSandbox so we can properly debug the issue? The `role="main"` on the main container div is what should resolve the issue, not the `role="presentation"`

Hi @tw15egan,here is the link for reduced version of the code on CodeSandbox -https://codesandbox.io/s/infallible-sid-pmcbw?file=/src/index.js

![image](https://user-images.githubusercontent.com/49453089/109220027-66443080-7786-11eb-9cd1-2ca587abc5b6.png)

As I had mentioned earlier, we are not applying any custom role here.We don't have a control over changing the current default `role="presentation"` to anything such as ` role="main"`. We are using the carbon components and this is set by Carbon.Let me know if you have any questions

**Comment by rsingh54** [comment]
> Going to close this out because it is working as intended when placed inside a proper region, like in our Storybook. If you feel like there is more for us to do, please feel free to reopen this with a link to a reproducible example so we can get this sorted out. Thanks!

I am unable to re -open this issue.Please reopen it.Sorry for late reply. If you still have any questions, we can get on a webex and discuss.Thanks!!

**Comment by tw15egan** [comment]
You need to set `role="main"` on a containing element, usually placed on the outermost wrapping body.

Here is an example, please take a look at line 8
https://codesandbox.io/s/bold-feather-hu2li?file=/src/index.js

**Comment by tw15egan** [comment]
<img width="1475" alt="Screen Shot 2021-02-25 at 1 56 56 PM" src="https://user-images.githubusercontent.com/11928039/109224836-55d58b00-7771-11eb-8507-9217ff973070.png">

**Comment by rsingh54** [comment]
> You need to set `role="main"` on a containing element, usually placed on the outermost wrapping body.
> 
> Here is an example, please take a look at line 8
> https://codesandbox.io/s/bold-feather-hu2li?file=/src/index.js

I tried setting  `role="main"`on the wrapping div.However, now, I am getting other accessibility violations related to multiple elements with role main should have unique labels and so on. 

![image](https://user-images.githubusercontent.com/49453089/109228598-12d6e000-7790-11eb-9051-4f511e56c6f7.png)

I also tried setting `aria-labelledby` and `aria-label` attributes as suggested in the solution.However, it leads to some other error messages.

I tried adding wrapping div here:
![image](https://user-images.githubusercontent.com/49453089/109228818-69dcb500-7790-11eb-970a-7f2491658c77.png)

**Comment by tw15egan** [comment]
You only need to have `role="main"` on one element, you're main application wrapping div that contains your Modal. The Modal just needs to be rendered inside of that element.

**Comment by rsingh54** [comment]
> You only need to have `role="main"` on one element, you're main application wrapping div that contains your Modal. The Modal just needs to be rendered inside of that element.

Yes, I had also tried applying `role="main"` on the wrapping div in main application which contains this Modal yesterday and tried again.Same issue.

![image](https://user-images.githubusercontent.com/49453089/109309959-f5e2f100-7811-11eb-8ff0-103f0069d8be.png)

The accessibility violation error still persists.
![image](https://user-images.githubusercontent.com/49453089/109310099-1ca12780-7812-11eb-9cb4-5f8ba79ad423.png)

After trying adding `aria-labelledby` attribute, I get this error -

![image](https://user-images.githubusercontent.com/49453089/109310705-e4e6af80-7812-11eb-964d-254763cee195.png)

**Comment by rsingh54** [comment]
Thanks for all the help.It works.The accessibility violation has been resolved after changing the placement of edit cluster modal inside the parent element.
